extensions/platform/src/core/artifacts/persistence.ts

```
@@ -0,0 +1,221 @@
+  } catch (error) {
+    operationFailed = true;
+    throw error;
+  } finally {
+    try {
+      if (temporaryCreated) await unlink(temporary);
+    } catch (error) {
+      if (
+        !operationFailed &&
+        (error as NodeJS.ErrnoException).code !== "ENOENT"
+      ) {
+        throw error;
```

Biome (2.5.7)

[error] Unsafe usage of 'throw'.

(lint/correctness/noUnsafeFinally)